ui/src/App.tsx

```diff
@@ -427,7 +427,12 @@ const AppComponent = () => {
   };
 
   const path = useLocation().pathname;
-  const currentEntry = entriesToDisplay.find(entry => path.startsWith(entry.path));
+
+  const currentEntry = entriesToDisplay.find(entry => path.startsWith(getBaseSegment(entry.path)));
+
+  function getBaseSegment(segment: string) {
+    return [segment.split('/')[0], segment.split('/')[1], segment.split('/')[2]].join('/');
+  }
```

Comment:
this makes it so `additional routes` fall under the same menuTitle as their parent segments. ex. Anything deeper than `app/custody` will have the `custody` menu title and so on